helix-term/src/commands.rs

```diff
@@ -3815,6 +3820,35 @@ pub mod insert {
         doc.apply(&transaction, view.id);
     }
 
+    pub fn insert_char_interactive(cx: &mut Context) {
+        let count = cx.count();
+
+        // need to wait for next key
+        cx.on_next_key(move |cx, event| {
+            match event {
+                KeyEvent {
+                    code: KeyCode::Char(ch),
+                    ..
+                } => {
+                    for _ in 0..count {
+                        insert::insert_char(cx, ch)
+                    }
+                }
+                KeyEvent {
+                    code: KeyCode::Enter,
+                    ..
+                } => {
+                    assert!(count == 1, "inserting multiple newlines not yet supported");
```

Comment:
Rather than a panic you can set an error in the statusline:

```rust
if count != 1 {
    cx.editor.set_error("inserting multiple newlines not yet supported");
}
```

Reply:
oh that's much better yes, thank you!